hooks/collect-metrics

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+# Load modules from $JUJU_CHARM_DIR/lib.
+import sys
+sys.path.append('lib')
+
+# Activate the virtualenv.
+from charms.layer.basic import activate_venv  # noqa: E402
+activate_venv()
+
+# Start the script as usual.
+import ssl  # noqa: E402
+import subprocess  # noqa: E402
+from urllib import request  # noqa: E402
+
+from firestealer import text_to_samples  # noqa: E402
+import yaml  # noqa: E402
+
+
+# Map metrics names to samples to be collected.
+METRICS = {
+    'containers_in_flight': 'jujushell_containers_in_flight',
+    'errors_count': 'jujushell_errors_count',
+    'requests_count': 'jujushell_requests_count',
+    'requests_duration': 'jujushell_requests_duration_sum',
+    'requests_in_flight': 'jujushell_requests_in_flight',
+}
+
+
+def main():
+    """Collect and add metrics to juju."""
+    url = service_url()
+    samples = retrieve_samples(url)
+    add_metrics(samples)
+
+
+def service_url():
+    """Retrieve the jujushell service URL by parsing its config file."""
+    # Note that, for reasons, the charmhelpers config object is not available
+    # in this hook.
+    with open('files/config.yaml') as f:
+        config = yaml.load(f)
+    return 'https://localhost:{}/metrics'.format(config['port'])
```

Comment:
https on localhost?

Reply:
Yes, the way we usually deploy jujushell is with let's encrypt, but this is a good point, we should switch to http based on the config. This can be done in a follow up I guess.